src/measure_theory/measure_space.lean

```diff
@@ -1823,3 +1823,85 @@ lemma metric.bounded.finite_measure {α : Type*} [metric_space α] [proper_space
   μ s < ⊤ :=
 (measure_mono subset_closure).trans_lt (metric.compact_iff_closed_bounded.2
   ⟨is_closed_closure, metric.bounded_closure_of_bounded hs⟩).finite_measure
+
+
+namespace measure_theory
+
+namespace measure
+
+/-- This is the equivalent of (a-b) ⊔ 0, if a and b were signed measures.
+Compare with ennreal.has_sub. 
+Specifically, note that if you have α = {1,2}, and  a {1} = 2, a {2} = 0, and 
+b {2} = 2, b {1} = 0, then (a - b) {1, 2} = 2. However, if a ≤ b, and
+b set.univ ≠ ⊤, then (a - b) + b = a. -/
+noncomputable instance has_sub {α:Type*}
+  [measurable_space α]:has_sub (measure_theory.measure α) := ⟨λa b, Inf {d | a ≤ d + b}⟩
+
+section measure_sub
+variables {α:Type*} [M:measurable_space α] {μ ν ν₁ ν₂:measure_theory.measure α}
+
+lemma sub_def:
+  (μ - ν) = Inf {d | μ ≤ d + ν} := rfl
+
+lemma le_zero_iff_eq:μ ≤ 0 ↔ μ = 0 :=
```

Comment:
This is not about `sub`. Please put it below lemma `measure.le_add_right`

Reply:
Moved it below zero_le.